app/controllers/subscriptions_controller.rb

```diff
@@ -96,11 +96,7 @@ def detect_plan_after_payment
   end
 
   def detect_user
-    user = if params[:user]
-             User.find_by(slug: params[:user])
-           else
-             current_user
-           end
+    user = params[:user] ? User.find_by(slug: params[:user]) : current_user
     User.find(user.id)
```

Comment:
is this redundant now? could we even change to:

```rb
return User.find_by slug: params[:user] if params[:user]
current_user
```

or even

```rb
User.find_by(slug: params[:user] || current_user.slug)
```

sorry, I've been gripped by refactoring madness!  that last one is silly - the two liner above probably best ...